create_primers.py

```diff
@@ -0,0 +1,261 @@
+
+Edited by Caelan Radford in 2020, from the CodonTilingPrimers code written by Jesse Bloom in 2013 and edited by Adam Dingens Nov 2015 to generate primers of differing lengths to all have a Tm of ~60C.
+
+This script first makes an ORIGINAL primer of specified length (default 37 bps).
+If the ORIGINAL primer has a Tm of greater than MaxPrimerTm, then nucleotides are trimmed off one by one (first 5', then 3', then 5' etc) until the Tm is less than MaxPrimerTm. Note that this could be over a degree less than the MaxPrimerTm.
+If the ORIGINAL primer has a Tm less than MinPrimerTm, then nucelotides are added one by one (first 3', then 5', then 3' etc) until the Tm is over MinPrimerTm. Note that this could be over a degree more than the MinPrimerTm
+If the ORIGINAL primer has a Tm of less than MaxPrimerTm but greater than MinPrimerTm, it is not altered.
+The primers are constrained to be between MinPrimerlength and MaxPrimerLength bps long. The Tm of some MaxPrimerLength primers may not be > MinPrimerTemp, and the Tm of some MinPrimerLength primers may not be < MaxPrimerTm.
+
+For command line arguments, run::
+
+    python create_primers.py -h
+
+The  Tm_NN command of the MeltingTemp Module of Biopython (http://biopython.org/DIST/docs/api/Bio.SeqUtils.MeltingTemp-module.html) is used to calculate Tm of primers.
+This calculation is based on nearest neighbor thermodynamics. nucelotides labeled N are given average values in the Tm calculation.
+It is possible to vary salt concentration and other addatives if needed."""
+
+
+import os
+import sys
+import math
+import re
+import argparse
+import pandas as pd
+from Bio.SeqUtils import MeltingTemp as mt
+from Bio.Seq import Seq
+
+
+def Parser():
+    """Returns command line parser."""
+    parser = argparse.ArgumentParser(
+            description='Script by Caelan Radford, Adam Dingens, and Jesse Bloom to design specified codon tiling primers with specific melting temperatures.',
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            )
+
+    parser.add_argument('sequencefile', help="the name of a file giving the sequence for which we are designing the primers. This file should only contain the sequence, and should not have any headers or other content. For the sequence, make the 5' and 3' ends that you do not want to mutate in lower case. Make the portion of the coding sequence that you want to tile with ambiguous codons in upper case. Typically, for example, the first site you mutate would be the initial start codon, so the first *ATG* would be the first upper case letter. You must have at least *(startprimerlength - 3) / 2* nucleotides in lower case at each end of the upper case sequence that you are mutating. This is because at least this much flanking sequence is needed to design primers of the indicated length; more sequence may be required if the primer at either end is extended beyond the startprimerlength.")
```

Comment:
Also please continue these strings onto new lines to maintain line length <= 79 characters.